### ISSUE TITLE
Crypt deprecation warning from passlib

### ISSUE THREAD

**Issue by gpkc** [fastapi-users/fastapi-users#1325]
## Describe the bug

Using passlib, which internally imports `crypt`, now gives a deprecation warning:

```
.venv/lib/python3.11/site-packages/passlib/utils/__init__.py:854
DeprecationWarning: 'crypt' is deprecated and slated for removal in Python 3.13
    from crypt import crypt as _crypt
```

**Comment by gpkc** [comment]
Found related issues:
https://foss.heptapod.net/python-libs/passlib/-/issues/148
https://foss.heptapod.net/python-libs/passlib/-/issues/187

https://github.com/fastapi-users/fastapi-users/issues/1301

**Comment by frankie567** [comment]
Indeed, but as we mentioned in #1301, there is nothing much we can do until `passlib` is updated 😓

**Comment by bkis** [comment]
As it seems quite unlikely passlib will receive any updates (it has been unmaintained for some years now), would it be possible to [use bcrypt directly](https://github.com/pyca/bcrypt/issues/684#issuecomment-1902590553) instead of relying on passlib? Or evalute possible alternatives?

**Comment by realitix** [comment]
Indeed it doesn't seem to be a good idea to rely on passlib anymore.

**Comment by JimScope** [comment]
Perhaps this option could be considered? https://github.com/fastapi-users/fastapi-users/pull/1345

**Comment by digitalkaoz** [comment]
passlib will never be fixed unless fixed there or forked

**Comment by hasB4K** [comment]
It seems that @frankie567, the main maintainer of this projet, decided to create a passlib alternative of its own 3 weeks ago.
https://github.com/frankie567/pwdlib

However, I don't know if he has a plan to replace passlib in fastapi-users yet. If someone wants to have the PasswordHelper code in the meantime, here it is:


```python

import secrets
import string
import pwdlib
from fastapi_users.password import PasswordHelperProtocol


class PasswordHelper(PasswordHelperProtocol):
    def __init__(self, context=None) -> None:
        self.context = context
        if self.context is None:
            self.context = pwdlib.PasswordHash.recommended()

    def verify_and_update(self, plain_password: str, hashed_password: str) -> Tuple[bool, str]:
        return self.context.verify_and_update(password=plain_password, hash=hashed_password)

    def hash(self, password: str) -> str:
        return self.context.hash(password)

    @staticmethod
    def _generate_password(length):
        alphabet = string.ascii_letters + string.digits
        password = ''.join(secrets.choice(alphabet) for _ in range(length))
        return password

    def generate(self) -> str:
        return self._generate_password(20)
```

**Comment by frankie567** [comment]
Hi @hasB4K 👋 

That's indeed the plan, just didn't have time to tackle it yet 😊

**Comment by hasB4K** [comment]
Like @MatthewScholefield said here https://github.com/fastapi-users/fastapi-users/issues/1301#issuecomment-1987579064, it seems that there is some news on the passlib side [here](https://foss.heptapod.net/python-libs/passlib/-/issues/187#note_346669).

**Comment by frankie567** [comment]
Indeed, but I think I'll still move forward `pwdlib` anyway. People will still be able to go back to passlib using a custom password helper if they want to.

**Comment by bkis** [comment]
I understand and I'm looking forward to what you're building there. If the relevant issues with passlib get fixed before your own lib is ready, it would still be nice if the passlib update finds its way into fastapi-users in the meantime. Thank you for all your work!

**Comment by frankie567** [comment]
I made the move and this is now fixed as of [`v13.0.0`](https://github.com/fastapi-users/fastapi-users/releases/tag/v13.0.0).

As I mention above, if you still want to use `passlib`, you can do so by implementing a custom `PasswordHelper`: https://fastapi-users.github.io/fastapi-users/latest/configuration/password-hash/#full-customization
